web/src/components/ReservationDatePicker/ReservationDatePicker.tsx

```diff
@@ -0,0 +1,108 @@
+import { Heading } from '@buildcities/build-ui.components.all'
+import 'react-dates/initialize'
+declare let window: any
+import {
+  DateRangePicker,
+  OrientationShape,
+  FocusedInputShape,
+} from 'react-dates'
+import { useEffect, useState } from 'react'
+import * as moment from 'moment'
+import 'react-dates/lib/css/_datepicker.css'
+
+// Move to presets.ts in relevant folder later
+const DATES_TEXT = 'Select dates'
+const CHECK_IN_TEXT = 'Check in'
+const CHECK_OUT_TEXT = 'Check out'
+
+const ReservationDatePicker = () => {
+  const [endDate, setEndDate] = useState<moment.Moment | null>(null)
+  const [startDate, setStartDate] = useState<moment.Moment | null>(null)
+  const [focus, setFocus] = useState<FocusedInputShape | null>(null)
+  const [open, setOpen] = useState<boolean>(false)
+  const [orientation, setOrientation] = useState<FocusedInputShape>(
+    'horizontal' as FocusedInputShape
+  )
+
+  const handleDateChange = (startDate, endDate) => {
+    setEndDate(endDate)
+    setStartDate(startDate)
+  }
+
+  const applyStyles = () => {
+    setOrientation(
+      (isMobile() ? 'vertical' : 'horizontal') as FocusedInputShape
```

Comment:
could this be achieved with the responsive classes? there is also a new feature in tailwind3 that supports orientation (https://tailwindcss.com/blog/tailwindcss-v3). I think its better we pack styling and design to css engine, performance is also better because the pre-processor will do the heavy lifting at compile time rather than this at runtime